subsys/bluetooth/services/ras/rrsp/ras_rd_buffer.c

```diff
@@ -150,15 +150,21 @@ static struct ras_rd_buffer *rd_buffer_alloc(struct bt_conn *conn, uint16_t rang
 	return NULL;
 }
 
-static void cs_procedure_enabled(struct bt_conn *conn,
-				 struct bt_conn_le_cs_procedure_enable_complete *params)
+static void cs_procedure_enable(struct bt_conn *conn,
+				uint8_t status,
+				struct bt_conn_le_cs_procedure_enable_complete *params)
 {
 	uint8_t conn_index = bt_conn_index(conn);
 
 	__ASSERT_NO_MSG(conn_index < ARRAY_SIZE(tx_power_cache));
 
-	tx_power_cache[conn_index] = params->selected_tx_power;
-	drop_procedure_counter[conn_index] = DROP_PROCEDURE_COUNTER_EMPTY;
+	if (status == BT_HCI_ERR_SUCCESS) {
+		tx_power_cache[conn_index] = params->selected_tx_power;
+		drop_procedure_counter[conn_index] = DROP_PROCEDURE_COUNTER_EMPTY;
+	} else {
```

Comment:
I think we can remove the else-clause here actually. This callback is registered in addition to the app's, so in case of failure this would be logged multiple times